nservicebus/upgrades/5to6-moving-away-from-ibus.md

```diff
@@ -7,28 +7,38 @@ Starting with Version 6, the `IBus` interface has been deprecated.
 
 The `IBus` interface was one of the key interfaces when using previous versions of NServiceBus. It provided access to many of the operations like sending messages, subscribing to events, and manipulating headers. The `IBus` interface was available through dependency injection in message handlers, any custom components that were registered in the container and through other mechanisms like the saga base class.
 
+
+## Reasons for deprecating IBus
+
+The `IBus` interface contained numerous methods and properties. However, not all of them are valid in all scenarios where an instance of `IBus` was exposed and this can lead to confusion. For example, the methods `Reply` and `ForwardCurrentMessageTo` are always available on the `IBus` interface, but they are only relevant to use it in the context of handling an incoming message. Using them in other scenarios will throw an exception.
+
+For users who are new to the NServiceBus API, the fact that `IBus` was available in message handlers using Dependency Injection wasn't obvious, especially when trying to send or publish messages from inside message handlers. Rather than using dependency injection, a `IMessageHandlerContext` parameter is now available in all the message handlers. This parameter will expose all of the appropriate actions to the message handler. Methods that aren't applicable will not be available making the API simpler.
```

Comment:
There is also a technical reason. Previously IBud impl was bound to a per request scope which sometimes was implemented as PerThreadScope. This doesn't work with async see my floating state post. Should we mention it?